src/python/WMCore/MicroService/Unified/Common.py

```diff
@@ -146,8 +148,11 @@ def getBlockReplicasAndSize(datasets, phedexUrl, group):
         for item in rows['phedex']['block']:
             block = {item['name']: {'blockSize': item['bytes'], 'locations': []}}
             for repli in item['replica']:
-                if repli['complete'] == 'y' and repli['subscribed'] == 'y' and repli['group'] == group:
-                    block[item['name']]['locations'].append(repli['node'])
+                if repli['complete'] == 'y' and repli['subscribed'] == 'y':
+                    if not group:
+                        block[item['name']]['locations'].append(repli['node'])
+                    elif repli['group'] == group:
```

Comment:
what will happen if group is not `None` but also does not match of replica group?

Reply:
In that case locations will remain an empty list (L149). Which means, MSTransferor must make a data subscription later in the execution.